plugins/org.eclipse.elk.core/src/org/eclipse/elk/core/LayoutConfigurator.java

```diff
@@ -218,16 +219,20 @@ public void visit(final ElkGraphElement element) {
     @SuppressWarnings("unchecked")
     protected void applyProperties(final ElkGraphElement element, final IPropertyHolder properties) {
         if (properties != null) {
-            if (!optionFilters.isEmpty()) {
-                for (Map.Entry<IProperty<?>, Object> entry : properties.getAllProperties().entrySet()) {
-                    boolean accept = optionFilters.stream()
-                            .allMatch(filter -> filter.accept(element, entry.getKey()));
-                    if (accept) {
-                        element.setProperty((IProperty<Object>) entry.getKey(), entry.getValue());
+            List<IOptionFilter> filters = getFilters();
+            for (Map.Entry<IProperty<?>, Object> entry : properties.getAllProperties().entrySet()) {
+                boolean accept = filters.stream()
+                        .allMatch(filter -> filter.accept(element, entry.getKey()));
+                if (accept) {
+                    Object value = entry.getValue();
+                    if (value instanceof Cloneable) {
+                        Object clone = ElkReflect.clone(value);
```

Comment:
Can this potentially silently fail? I believe `ElkReflect` raises an exception if it fails to clone, right?

Reply:
The documentation says is returns `null` if it fails.

Reply:
Hence, if a clone fails, it will set a potentially existing property to `null`, effectively removing it? 

Not that a clone is supposed to fail here in the first place, I'm just wondering if this introduces another potential issue that is similarly hard to track down.